library/general/src/lib/cfa/nsswitch.rb

```diff
@@ -0,0 +1,273 @@
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cfa/base_model"
+require "yast2/target_file"
+
+Yast.import "FileUtils"
+
+module CFA
+  # Model to handle the Name Service Switch configuration file
+  #
+  # @note Since nsswitch.conf does not allow to overrides single entries, the content will be
+  #   read from /usr/etc/nsswitch.conf only when /etc/nsswitch.conf file does not exist yet.
+  #   In any case, changes will be always written to /etc/nsswitch.conf.
+  #
+  # @example Reading service specifications of a database
+  #   file = Nsswitch.new
+  #   file.load
+  #   file.services_for("hosts") #=> ["db", "files"]
+  #
+  # @example Writing service specifications for a database
+  #   file = Nsswitch.new
+  #   file.load
+  #   file.update_entry("hosts", ["db", "files"])
+  #   file.save
+  #
+  # @example Deleting a database entry
+  #   file = Nsswitch.new
+  #   file.load
+  #   file.delete_entry("hosts")
+  #   file.save
+  #
+  # @example Loading shortcut
+  #   file = Nsswitch.load
+  #   file.entries #=> ["passwd", "group", "shadow", "hosts", "networks", "aliases"]
+  class Nsswitch < BaseModel
+    include Yast::Logger
+
+    # Path to local configuration file
+    LOCAL_PATH = "/etc/nsswitch.conf".freeze
+    private_constant :LOCAL_PATH
+
+    # Path to vendor configuration file
+    VENDOR_PATH = "/usr/etc/nsswitch.conf".freeze
+    private_constant :VENDOR_PATH
+
+    class << self
+      # Instantiates and loads the file
+      #
+      # This method is basically a shortcut to instantiate and load the content in just one call.
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Nsswitch] File with the already loaded content
+      def load(file_handler: Yast::TargetFile)
+        new(file_handler: file_handler).tap(&:load)
+      end
+    end
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_handler: Yast::TargetFile)
+      super(AugeasParser.new("nsswitch.lns"), load_path, file_handler: file_handler)
+    end
+
+    # Loads the file content
+    #
+    # If the file does not exist, consider it as empty.
+    #
+    # @see CFA::BaseModel#load
+    def load
+      super
+      fix_collection_names(data)
+      @current_content = @parser.serialize(data)
+    rescue Errno::ENOENT # PATH does not exist yet
```

Comment:
well, this is a bit abuse of knowing file_handler, but currently I do not know better way to check by asking file handler if file exists. It is on my todo list.